x-pack/plugins/infra/public/containers/waffle/nodes_to_wafflemap.ts

```diff
@@ -84,10 +84,12 @@ function findOrCreateGroupWithGroups(
 }
 
 function createWaffleMapNode(node: InfraNode): InfraWaffleMapNode {
+  const nodePathItem = last(node.path);
   return {
     id: node.path.map(p => p.value).join('/'),
     path: node.path,
-    name: last(node.path).value,
+    nodeId: nodePathItem.value,
+    name: nodePathItem.displayName || nodePathItem.value,
```

Comment:
If we already fell back to the `id` on the server side, this fallback would be unnecessary.